docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -82,30 +82,158 @@ a :ref:`connector-page-sink-provider`.
+--------------------------
 
 Given a split and a list of columns, the record set provider is
 responsible for delivering data to the Trino execution engine.
+
 It creates a ``RecordSet``, which in turn creates a ``RecordCursor``
 that is used by Trino to read the column values for each row.
 
+The provided record set must only include requested columns
+in the correct order. If it returns more rows than necessary,
+the engine will filter them out unless the connector implements
+pushing down predicates, limit, or top-N, and informs the engine
+that this is not necessary.
+
+For simple connectors, where performance is not critical,
+the record set provider can return an instance of ``InMemoryRecordSet``.
+It can be built using lists of values for every row, which can be
+simpler than implementing a ``RecordCursor``.
+
+A ``RecordCursor`` implementation needs to keep track of the current record.
+It return values for columns by a numerical position, in the data type matching
+the column definition in the table. When the engine is done reading the current
+record it will call ``advanceNextPosition`` on the cursor.
+
+Type mapping
+^^^^^^^^^^^^
+
+The built-in SQL data types use different Java types as carrier types.
+
+================================== =============
+SQL type                           Java type
+================================== =============
+``BOOLEAN``                        ``boolean``
+``TINYINT``                        ``long``
+``SMALLINT``                       ``long``
+``INTEGER``                        ``long``
+``BIGINT``                         ``long``
+``REAL``                           ``double``
+``DOUBLE``                         ``double``
+``DECIMAL``                        ``long`` for precision up to 19, inclusive;
+                                   ``Int128`` for precision greater than 19
+``VARCHAR``                        ``Slice``
+``CHAR``                           ``Slice``
+``VARBINARY``                      ``Slice``
+``JSON``                           ``Slice``
+``DATE``                           ``long``
+``TIME(P)``                        ``long``
+``TIME WITH TIME ZONE``            ``long`` for precision up to 9;
+                                   ``LongTimeWithTimeZone`` for precision
+                                   greater than 9
+``TIMESTAMP(P)``                   ``long`` for precision up to 6;
+                                   ``LongTimestamp`` for precision greater
+                                   than 6
+``TIMESTAMP(P) WITH TIME ZONE``    ``long`` for precision up to 3;
+                                   ``LongTimestampWithTimeZone`` for precision
+                                   greater than 3
+``INTERVAL YEAR TO MONTH``         ``long``
+``INTERVAL DAY TO SECOND``         ``long``
+``ARRAY``                          ``Block``
+``MAP``                            ``Block``
+``ROW``                            ``Block``
+``IPADDRESS``                      ``[]byte``
```

Comment:
byte[]

However, the carrier type for IPADDRESS and UUID is Slice

Reply:
Thanks! I used `InMemoryRecord` set too much :-)